### PR TITLE
Update react-storybook to latest version to fix iframe on docs site

### PR DESCRIPTION
**Pull request by halkeye**
**Description**
- Currently https://jenkinsci.github.io/jenkins-design-language/docs iframes the storybook pages. Unfortuatly due to an existing bug, the iframes don't work
- See https://github.com/storybooks/react-storybook/issues/555
- This has already been addressed in later versions of react-storybook, so this PR just updates the dependancy

**Submitter checklist**
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
cc @sophistifunk @cliffmeyers look ok?

**Comment by sophistifunk** [comment]
Does this build in our cloud? Github is telling me it failed, but the URL is bad.

**Comment by michaelneale** [comment]
@sophistifunk yeah it does - but a test instance (noting to do with dogfood) wallopped the status. 

@dragoonis if you push a change it shoudl trigger another build.

**Comment by michaelneale** [comment]
@dragoonis you have an invitation to be a collaborator on this repo so you can work in branches here too if that suits.

**Comment by cliffmeyers** [comment]
@michaelneale this PR doesn't show up on the PR's tab on Dogfood. Should it? Otherwise I'm not sure how to re-run the build.

**Comment by michaelneale** [comment]
I don't know either - convert it to a branch and it should rebuild.

**Comment by halkeye** [comment]
I think https://github.com/jenkinsci/jenkins-design-language/blob/master/site-publish.sh isn't part of the build anymore.

**Comment by sophistifunk** [comment]
> I think https://github.com/jenkinsci/jenkins-design-language/blob/master/site-publish.sh isn't part of the build anymore.

Never was, as it doesn't work for everybody.

**Comment by scherler** [comment]
@sophistifunk what is the problem. I fixed it around a month ago when i published the last time

**Comment by sophistifunk** [comment]
IIRC, it required a specific version of git that's newer than what you get on OSX.

**Comment by scherler** [comment]
@sophistifunk yeah I added:

```
git merge --no-edit -s ours remotes/webpages/gh-pages --allow-unrelated-histories
```

` --allow-unrelated-histories` is to allow that merge if you are on git 2.9+

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "react-dom": "15.3.2"
   },
   "devDependencies": {
-    "@kadira/storybook": "2.24.1",
+    "@kadira/storybook": "^2.35.3",
```
Comment (scherler) [inline package.json]: `"@kadira/storybook": "2.35.3"` please do not use `^`

**Comment by halkeye** [comment]
Doesn't yarn.lock handle that anyways? I can't fix this or on mobile so I'll try and get to it asap

File: package.json (hunk 1 above)
Comment (cliffmeyers) [inline package.json]: I was thinking, should we write a utility that validates package.json for semver best practices. My feeling is that deps and devDeps should use exact, but peerDeps should use caret operator. We could create a standalone util for use in JDL but also bake it into js-builder by default. Or maybe something comparable already exists, can look. WDYT @scherler ?

**Comment by michaelneale** [comment]
@halkeye yarn isn't in use yet (and probably better to be explicit, like you have been). 

@scherler package.json has been updated.

**Comment by michaelneale** [comment]
ping @scherler @sophistifunk can we merge this one in now?

**Comment by sophistifunk** [comment]
Definitely

**Comment by michaelneale** [comment]
nice one @halkeye !

*swish*

**Pull request merged**